handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -301,8 +299,14 @@ SSLEngineResult unwrap(SslHandler handler, ByteBuf in, int readerIndex, int len,
             }
 
             @Override
-            int calculateWrapBufferCapacity(SslHandler handler, int pendingBytes, int numComponents) {
-                return handler.engine.getSession().getPacketBufferSize();
+            ByteBuf allocateWrapBuffer(SslHandler handler, ByteBufAllocator allocator,
+                                       int pendingBytes, int numComponents) {
+                // As for the JDK SSLEngine we always need to allocate buffers of the size of 16kb
```

Comment:
NIT: It's technically over 16KB by default. Something around 16609. And can be configured to something different based on the context. Maybe say:

```
// As for the JDK SSLEngine we always need to allocate buffers of the size required by the SSLEngine (normally
// ~16KB). This is required even if the amount of data to encrypt is very small. Use heap buffers to reduce
```

Reply:
good idea... @tbrooks8 I wonder if you could also test / verify / benchmark this change as it seems you care about this :)

Reply:
This looks good to me and thanks!